x-pack/plugin/esql/compute/src/main/generated-src/org/elasticsearch/compute/data/IntVectorFixedBuilder.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.compute.data;
+
+import org.apache.lucene.util.RamUsageEstimator;
+
+/**
+ * Builder for {@link IntVector}s that never grows. Prefer this to
+ * {@link IntVectorBuilder} if you know the precise size up front because
+ * it's faster.
+ * This class is generated. Do not edit it.
+ */
+final class IntVectorFixedBuilder implements IntVector.FixedBuilder {
+    private final BlockFactory blockFactory;
+    private final int[] values;
+    /**
+     * The next byte to write into. {@code -1} means the vector has already
+     * been built.
+     */
+    private int i;
+
+    IntVectorFixedBuilder(int size, BlockFactory blockFactory) {
+        blockFactory.adjustBreaker(size(size), false);
+        this.blockFactory = blockFactory;
+        this.values = new int[size];
+    }
+
+    @Override
+    public IntVectorFixedBuilder appendInt(int value) {
+        values[i++] = value;
+        return this;
+    }
+
+    private static long size(int size) {
+        return size == 1
+            ? ConstantIntVector.RAM_BYTES_USED
+            : IntArrayVector.BASE_RAM_BYTES_USED + RamUsageEstimator.alignObjectSize(
+                (long) RamUsageEstimator.NUM_BYTES_ARRAY_HEADER + size * Integer.BYTES
```

Comment:
This is fine. Separately we should update `IntArrayVector::ramBytesEstimated`, and have it based on an array length, rather than an array - then it can be used here as well as for the built vector.  ( I think you made a similar comment on that original PR )